timemachine/fe/free_energy.py

```diff
@@ -1494,6 +1501,13 @@ def sample_replica(xvb: CoordsVelBox, state_idx: StateIdx) -> tuple[NDArray, NDA
             )
             assert frame.shape[0] == 1
 
+            if md_params.water_sampling_params is not None:
+                for mover in context.get_movers():
+                    if isinstance(mover, WATER_SAMPLER_MOVERS):
```

Comment:
nit: would it be simpler to retain a handle to the water sampler mover above to avoid the need to search for it again here?